src/state/dataSelectors.ts

```diff
@@ -0,0 +1,44 @@
+import { createSelector } from '@reduxjs/toolkit';
+import { RechartsRootState } from './store';
+import { ChartData, ChartDataState } from './chartDataSlice';
+import { DataKey } from '../util/types';
+import { getValueByDataKey } from '../util/ChartUtils';
+
+/**
+ * This is a "cheap" selector - it returns the data but doesn't iterate them, so it is not sensitive on the array length.
```

Comment:
what do you mean by 

> it is not sensitive on the array length

This is just a simple selector, it is not ever going to be sensitive to anything?